lib/wallets/crypto_currency/coins/monero.dart

```diff
@@ -122,6 +144,105 @@ class Monero extends CryptonoteCurrency {
+    if (walletData.txids != null) {
+      final wallet = await Wallet.create(
+        walletInfo: WalletInfo.createNew(
+          coin: walletData.coin,
+          name: "${walletData.coin.prettyName} Paper Wallet",
+          restoreHeight: walletData.height ?? 0,
+          otherDataJsonString: walletData.toJson(),
+        ),
+        mainDB: ref.read(mainDBProvider),
+        secureStorageInterface: ref.read(secureStoreProvider),
+        nodeService: ref.read(nodeServiceChangeNotifierProvider),
+        prefs: ref.read(prefsChangeNotifierProvider),
+        mnemonicPassphrase: null,
+        mnemonic: walletData.seed,
+      );
+      await (wallet as MoneroWallet).init(isRestore: true);
+      await wallet.recover(isRescan: false);
+
+      // Scan the blocks with given txids
+      final primaryNode = defaultNode;
+      final daemonRpc = DaemonRpc(
+          IOClient(HttpClient()), "${primaryNode.host}:${primaryNode.port}");
+      final txs = await daemonRpc.postToEndpoint("/get_transactions", {
+        "txs_hashes": walletData.txids,
+        "decode_as_json": true,
+      });
+      for (final tx in txs["txs"] as List) {
+        wallet.onSyncingUpdate(syncHeight: tx["block_height"] as int,
+            nodeHeight: wallet.currentKnownChainHeight);
+        await wallet.waitForBalanceChange();
+      }
+
+      // Set the sync height to the current known chain height - make the wallet synced
+      wallet.onSyncingUpdate(syncHeight: wallet.currentKnownChainHeight,
+          nodeHeight: wallet.currentKnownChainHeight);
+
+      // Create the new wallet info
+      final newWallet = await Wallet.create(
+        walletInfo: WalletInfo.createNew(
+          coin: walletData.coin,
+          name: "${walletData.coin.prettyName} Gift Wallet ${walletData.address != null ? '(${walletData.address!.substring(walletData.address!.length - 4)})' : ''}",
+        ),
+        mainDB: ref.read(mainDBProvider),
+        secureStorageInterface: ref.read(secureStoreProvider),
+        nodeService: ref.read(nodeServiceChangeNotifierProvider),
+        prefs: ref.read(prefsChangeNotifierProvider),
+        mnemonic: null,
+        mnemonicPassphrase: null,
+        privateKey: null,
+      );
+      await (newWallet as MoneroWallet).init(wordCount: 16);
+      await newWallet.info.setMnemonicVerified(isar: ref
+          .read(mainDBProvider)
+          .isar);
+      ref.read(pWallets).addWallet(newWallet);
+
+      await newWallet.open();
+      await newWallet.generateNewReceivingAddress();
+
+      // Send the balance to the new wallet
+      final txData = await wallet.prepareSend(txData: TxData(
+        recipients: [
+          (address: (await newWallet.getCurrentReceivingAddress())!.value, amount: (await wallet.totalBalance), isChange: false)
+        ],
+        feeRateType: FeeRateType.average,
+      ));
+      await wallet.confirmSend(txData: txData);
```

Comment:
What happens to wallet after this send?  This temporary wallet should at least be closed and probably deleted.  That should also require user input similar to the previous recommendation to use ConfirmTransactionView ... perhaps they can be combined or you can just ask for confirmation to send (sweep) and then present a separate dialog box asking if the swept wallet should be deleted